src/spec.rs

```diff
@@ -1,9 +1,33 @@
 use std::collections::HashMap;
 
 use obj_ser;
-use object_hash::get_base64_hash;
+use object_hash::{get_base64_hash, get_chash};
 use serde_json::Value;
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Login {
+    pub challenge: Option<String>,
+    pub pubkey: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub signature: Option<String>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TempPubkey {
+    pub pubkey: Option<String>,
```

Comment:
the same, if all fields are Option, then it can always be matched, and not fall back to `Other` variant.